web/src/types/api.ts

```diff
@@ -111,3 +101,27 @@ export interface Run {
 }
 
 export type RunState = 'NEW' | 'COMPLETED' | 'FAILED' | 'ABORTED'
+
+export interface SearchResult {
+  name: string
+  namespace: string
+  nodeId: string
+  type: JobOrDataset
+  updatedAt: string
+}
+
+export type GroupedSearch = { group: string } & SearchResult
+
+export interface LineageGraph {
+  graph: LineageNode[]
+}
+
+export interface Search {
+  totalCount: number
+  results: SearchResult[]
+}
+
+export interface GroupedSearchResult {
+  results: Map<string, GroupedSearch[]>
```

Comment:
I don't think we need that since we mix them in our rendering. We can also do some pretty inexpensive client-side grouping if we wish to evolve on the product side.